src/platform/debugger/multiplexingDebugService.ts

```diff
@@ -78,21 +84,41 @@ export class MultiplexingDebugService implements IJupyterDebugService {
         return this.activeService.onDidChangeBreakpoints;
     }
     public get onBreakpointHit(): Event<void> {
+        if (!this.jupyterDebugService) {
+            throw new Error('No jupyter debugger service');
+        }
         return this.jupyterDebugService.onBreakpointHit;
     }
     public startRunByLine(config: DebugConfiguration): Thenable<boolean> {
         this.lastStartedService = this.jupyterDebugService;
+        if (!this.jupyterDebugService) {
+            throw new Error('No run by line debugger service');
```

Comment:
This is just a fault of our own naming issues. But I would probably just say "No Jupyter Debug Service" like you did on the previous Error just a few lines above. We inject it with the Run by line identifier, but it's really the Jupyter debug service missing. Or maybe "Couldn't start run by line. Jupyter Debug Service missing".